Add support for Windows auth in mssql_resource

nsdavidson opened the pull request:
This adds supports for connecting to MS SQL instances using Window
authentication rather than SQL authentication.  By leaving either the
user or password parameters blank causes the sqlcmd to leave off the -U
and -P params.  This will cause sqlcmd to authenticate as the current
Windows user.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Add support for Windows auth in mssql_resource

This adds supports for connecting to MS SQL instances using Window
authentication rather than SQL authentication.  By leaving either the
user or password parameters blank causes the sqlcmd to leave off the -U
and -P params.  This will cause sqlcmd to authenticate as the current
Windows user.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Add support for Windows auth in mssql_resource

This adds supports for connecting to MS SQL instances using Window
authentication rather than SQL authentication.  By leaving either the
user or password parameters blank causes the sqlcmd to leave off the -U
and -P params.  This will cause sqlcmd to authenticate as the current
Windows user.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

aaronlippold reviewed (changes requested):
there is a small typo in the ##Syntax section. `describe mysql_session(` should be `describe mssql_session(` correct? 

Also, same issue below that we ran into with the postgres and mysql session resources. 

```Open a Command Prompt window, and type sqlcmd -S myServer\instanceName. Replace myServer\instanceName with the name of the computer and the instance of SQL Server that you want to connect to.``` https://technet.microsoft.com/en-us/library/ms166559(v=sql.105).aspx.

I would suggest that if sqlcmd allows the definition of a host an 'instance'/'tenant' - we should provide the ability for the user to specify that but default them to either nil or localhost if that is the 80%-90% use case but we shouldn't assume they will also be the case.

Commit:
Refactors to support non-default host/instance

Changed the initialize method to take an option hash so that we can
take optional host and instance parameters in addition to credentials.

Updated docs to reflect changes.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Resolving conflicts

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Resolving rubocop issues

Signed-off-by: Nolan Davidson <ndavidson@chef.io>
